### PR TITLE
Ordering inputFiles

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] spike idea in README
- [x] tests
- [x] implementation
- [x] decide on public API method name
- [ ] get someone elses 👀  on this

cc @chriseppstein

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,50 @@ The structure of `output.js` will be as follows:
...
+observed. Do not add files not already matched by `inputFiles` globs, as those
+files will not be considered for cache eviction on the next rebuild.
+
+```js
+var globalMatcher  = new Minimatch('*-global.css', { matchBase: true });
+var extMatcher     = new Minimatch('*-ext.css', { matchBase: true });
+
+var node = concat(node, {
+  inputFiles: ['**/*.css'],
+  someMethodName(files) {
```
Comment (stefanpenner) [inline README.md]: not sure what to call this yet

File: README.md [hunk 2]
```diff
@@ -43,6 +43,50 @@ The structure of `output.js` will be as follows:
 // - footer
 ```
 
+##### Ordering inputFiles
+
+Although when concatenating some types of files, AMD modules for example
```
Comment (stefanpenner) [inline README.md]: this words will need rework

File: README.md [hunk 3]
```diff
@@ -43,6 +43,50 @@ The structure of `output.js` will be as follows:
...
+      } else {
+        core.push(f);
+      }
+    });
+
+    return global.concat(core, ext);
+  }
+});
+```
+
```
Comment (stefanpenner) [inline README.md]: explain output in example?

File: README.md (hunk 1 above)
Comment (stefanpenner) [inline README.md]: this should actually most likely just be a simple comparator

File: README.md [hunk 4]
```diff
@@ -43,6 +43,48 @@ The structure of `output.js` will be as follows:
...
+A common alternative is to sort these files based on the matcher precedent.
+
+Given:
+
+```
+inputFiles: ['foo/*.global.css', 'foo/*.overides.css']
+```
+
+One may want, want all global css files to proceed overide css files, while
```
Comment (hjdivad) [inline README.md]: typo `want, want`

**Comment by micky2be** [comment]
That really important to keep the order, please merge this asap

**Comment by codynguyen** [comment]
Any chance this will be merged soon? I'd be glad to help but don't know how

**Comment by stefanpenner** [comment]
I need to remember why this wasn't merged.. 
@codynguyen is your use-case specifically the pluggable comparator?

**Comment by codynguyen** [comment]
@stefanpenner yes this pluggable comparator is exactly what our project need

**Comment by stefanpenner** [comment]
@codynguyen makes sense, if you can rebase your PR I'll gladly merge it. It seems good, I think I didn't merge this because I simply didn't circle back.

**Pull request closed without merging**